# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1552632986 0
#      Fri Mar 15 06:56:26 2019 +0000
# Node ID 9c8ebafba517040f4931d3eddd513aa84ae1315f
# Parent  e0111653adcf0422e855b5698fb160e233c836bf
avoid some overloaded virtual warnings (bug #55741)

* gl-render.h (opengl_renderer::draw (const Matrix&, bool)): Don't
declare virtual since no derived class needs to override it.

diff --git a/libinterp/corefcn/gl-render.h b/libinterp/corefcn/gl-render.h
--- a/libinterp/corefcn/gl-render.h
+++ b/libinterp/corefcn/gl-render.h
@@ -47,17 +47,20 @@ namespace octave
     opengl_renderer& operator = (const opengl_renderer&) = delete;
 
     virtual ~opengl_renderer (void) = default;
 
     opengl_functions& get_opengl_functions (void) const { return m_glfcns; }
 
     virtual void draw (const graphics_object& go, bool toplevel = true);
 
-    virtual void draw (const Matrix& hlist, bool toplevel = false)
+    // The following version of the draw method is not declared virtual
+    // because no derived class overrides it.
+
+    void draw (const Matrix& hlist, bool toplevel = false)
     {
       int len = hlist.numel ();
 
       for (int i = len-1; i >= 0; i--)
         {
           graphics_object obj = gh_manager::get_object (hlist(i));
 
           if (obj)
